app/src/main/java/com/example/concordia_campus_guide/Helper/PathFinder.java

```diff
@@ -57,21 +58,24 @@ public PathFinder(final Context context, final RoomModel source, final RoomModel
         return walkingPointNodesMap;
     }
 
-    protected WalkingPoint getWalkingPointCorrespondingToRoom(final RoomModel room,
-            final List<WalkingPoint> walkingPointList) {
-        final Coordinates coordinates1 = new Coordinates(room.getCenterCoordinates()[0],
-                room.getCenterCoordinates()[1]);
-        final WalkingPoint wantedPoint = new WalkingPoint(coordinates1, room.getFloorCode(), null, null);
+    protected WalkingPoint getWalkingPointCorrespondingToPlace(final Place place) {
+        String placeCode;
+        List<WalkingPoint> pointList = null;
+        if (place instanceof Building) {
+            placeCode = ((Building) place).getBuildingCode();
+            pointList = AppDatabase.getInstance(context).walkingPointDao().getAllWalkingPointsFromPlaceCode(placeCode);
+        } else if (place instanceof RoomModel) {
+            placeCode = ((RoomModel) place).getRoomCode();
+            pointList = AppDatabase.getInstance(context).walkingPointDao().getAllWalkingPointsFromPlaceCode(placeCode);
```

Comment:
each of the extended classes (roomModel, Floor & building ) have different code attributes but they are all the same type and have the same purpose. Why don't we create a placeCode  in the place class  ? Then we won't need to have these if statements. What do you think ?Also, @thomasgauvin @tarekait1996  what do you guys think too?

Reply:
Yes we should investigate this possibility

Reply:
I will create an improvement task

Reply:
https://github.com/Concordia-Campus-Guide/Concordia-Campus-Guide/issues/129

Reply:
I see they also introduced a place code attribute, which I don't exactly understand how its being used. If it's being used the way you speak of above @Hambrsoom , we should do this as part of this story

Reply:
As you mentioned before, this is not something related to this user story specifically. placeCode is like an id for the each type of place. That's why i thought we could have put it in the father class instead of having it in each sub class

Reply:
yes the placCode maps to either the building code, or the roomCodes of the places in order to get the place corresponding to the walkingpoint.